src/MicroOrm.Dapper.Repositories/ReadOnlyDapperRepository.cs

```diff
@@ -36,10 +39,18 @@ public ReadOnlyDapperRepository(IDbConnection connection, ISqlGenerator<TEntity>
         }
 
         /// <inheritdoc />
-        public IDbConnection Connection { get; set; }
+        public IDbConnection Connection
+        {
+            get => _connection ?? throw new ArgumentNullException(nameof(_connection));
+            set => _connection = value;
+        }
 
         /// <inheritdoc />
-        public FilterData FilterData { get; set; }
+        public FilterData FilterData
```

Comment:
`FilterData` initialized in constructor, should be

```c#
public FilterData FilterData { get; set; } 
```

Reply:
`FilterData` will be set null on `Dispose()`, throw `ObjectDisposedException` now.